### PR TITLE
Add JJBB jobs to .ci

### PR DESCRIPTION
**Pull request by jonahbull**
This adds the Jenkins jobs definitions to the repo so they can be
managed and updated here.

### CODE REVIEW

**Comment by jonahbull** [comment]
Here you go, @delvedor, hopefully this works out better than my last attempt! 😅

File: .ci/jobs/elastic+elasticsearch-js+next.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+- job:
```
Comment (delvedor) [inline .ci/jobs/elastic+elasticsearch-js+next.yml]: What is the trigger for this job?

Comment (jonahbull) [reply .ci/jobs/elastic+elasticsearch-js+next.yml]: It'll inherit the default `github` trigger from `defaults.yml` when JJBB merges the jobs and defaults.

Comment (delvedor) [reply .ci/jobs/elastic+elasticsearch-js+next.yml]: Great!
And the `github` trigger means that it will run a build for every commit in the branch?

Comment (jonahbull) [reply .ci/jobs/elastic+elasticsearch-js+next.yml]: Yep, that's right.

These files will produce the same behavior as the jobs that currently exist on `clients-ci`.

**Pull request merged**